swarm/api/http/server.go

```diff
@@ -962,12 +944,24 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *Request) {
 	// check the root chunk exists by retrieving the file's size
 	if _, err := reader.Size(ctx, nil); err != nil {
 		getFileNotFound.Inc(1)
-		Respond(w, r, fmt.Sprintf("file not found %s: %s", r.uri, err), http.StatusNotFound)
+		RespondError(w, r, fmt.Sprintf("file not found %s: %s", uri, err), http.StatusNotFound)
+		sp.Finish()
+		return
+	}
+
+	buf, err := ioutil.ReadAll(newBufferedReadSeeker(reader, getFileBufferSize))
```

Comment:
Remove this please. move `sp.Finish()` in a defer statement, since we no longer measure the request within the cluster.